### ISSUE TITLE
Invalid argument: Instance of 'SpeechSynthesisEvent'

### ISSUE THREAD

**Issue by alanorder** [dlutton/flutter_tts#256]
## 💬 Questions and Help

I am trying to call flutter_tts after a page has finished building:

  @override
  void initState() {
    super.initState();
    WidgetsBinding.instance.addPostFrameCallback((_) {
      _speak('testing');
    });
  }
However, I obtained this error:

Invalid argument: Instance of 'SpeechSynthesisEvent'

Flutter_tts is working perfectly if called within an onPressed().

Could anyone help? Thanks a lot

**Comment by dlutton** [comment]
@alanorder can you provide info on what SpeechSynthesisEvent is?

**Comment by alanorder** [comment]
Dear Daniel,

SpeechSynthesisEvent is not defined in my project. I googled and I
believe it is defined in the Web Speech API:

https://developer.mozilla.org/en-US/docs/Web/API/SpeechSynthesisEvent

The *SpeechSynthesisEvent* interface of the Web Speech API
<https://developer.mozilla.org/en-US/docs/Web/API/Web_Speech_API> contains
information about the current state of SpeechSynthesisUtterance
<https://developer.mozilla.org/en-US/docs/Web/API/SpeechSynthesisUtterance>
objects
that have been processed in the speech service.

Hope this is useful.

**Comment by alanorder** [comment]
Just to supplement that my project is a web application.

**Comment by dlutton** [comment]
ah I didn't realize you were using this on web.  Where are you instantiating the FlutterTTS class? As long as that is happening before hand, it should work with `WidgetsBinding.instance.addPostFrameCallback((_) =>
_speak('testing');
});`

**Comment by alanorder** [comment]
I think I have found the reason. It has nothing to do with flutter_tts.

This is the beginning of my code which is the same as your example:

class _MyAppState extends State<MyApp> {
  late FlutterTts flutterTts;
  String? language;
  String? engine;
  double volume = 0.5;
  double pitch = 1.0;
  double rate = 0.5;
  bool isCurrentLanguageInstalled = false;

  String? _newVoiceText;
  int? _inputLength;

  TtsState ttsState = TtsState.stopped;

  get isPlaying => ttsState == TtsState.playing;
  get isStopped => ttsState == TtsState.stopped;
  get isPaused => ttsState == TtsState.paused;
  get isContinued => ttsState == TtsState.continued;

  bool get isIOS => !kIsWeb && Platform.isIOS;
  bool get isAndroid => !kIsWeb && Platform.isAndroid;
  bool get isWeb => kIsWeb;

  @override
  initState() {
    super.initState();
    initTts();
    WidgetsBinding.instance?.addPostFrameCallback((_) {
      _speakTest();
    });
  }

The reason I received the error was due to the Chrome Autoplay policy:

https://developers.google.com/web/updates/2017/09/autoplay-policy-changes

As my application called flutter_tts once after building the page, before any user interaction, this is prohibited by Chrome.

I managed to make it works in Safari, which has an easy setting to allow autoplay.

Still trying to do the same for Chrome.

Thanks for you response and help.
